addons/website_event/controllers/main.py

```diff
@@ -415,14 +417,44 @@ def get_formated_date(self, event):
         month = babel.dates.get_month_names('abbreviated', locale=get_lang(event.env).code)[start_date.month]
         return ('%s %s%s') % (month, start_date.strftime("%e"), (end_date != start_date and ("-" + end_date.strftime("%e")) or ""))
 
-    def _extract_searched_event_tags(self, searches):
+    def _extract_searched_event_tags(self, searches, slug_tags):
         tags = request.env['event.tag']
-        if searches.get('tags'):
-            try:
-                tag_ids = literal_eval(searches['tags'])
-            except:
-                pass
-            else:
-                # perform a search to filter on existing / valid tags implicitely + apply rules on color
-                tags = request.env['event.tag'].search([('id', 'in', tag_ids)])
+        if slug_tags:
+            tags = self._event_search_tags_slug(slug_tags)
+        elif 'tags' in searches:
+            tags = self._event_search_tags_ids(searches['tags'])
         return tags
+
+    def _slugify_tags(self, tag_ids, toggle_tag_id=None):
+        """ Prepares a comma separated slugified tags for the sake of readable URLs.
+        :param toggle_tag_id: add the tag being clicked (current_tag) to the already
+          selected tags (tag_ids) as well as in URL; if tag is already selected
+          by the user it is removed from the selected tags (and so from the URL);
+        """
+        tag_ids = list(tag_ids)
+        if toggle_tag_id and toggle_tag_id in tag_ids:
+            tag_ids.remove(toggle_tag_id)
+        elif toggle_tag_id:
+            tag_ids.append(toggle_tag_id)
+
+        return ','.join(request.env['ir.http']._slug(tag_id) for tag_id in request.env['event.tag'].browse(tag_ids)) or ''
+
+    def _event_search_tags_ids(self, search_tags):
+        """ Input: %5B4%5D """
+        EventTag = request.env['event.tag']
+        try:
+            tag_ids = literal_eval(search_tags or '')
+        except Exception: # noqa: BLE001
+            return EventTag
+
+        return EventTag.search([('id', 'in', tag_ids)]) if tag_ids else EventTag
+
+    def _event_search_tags_slug(self, search_tags):
+        """ Input: event-1,event-2 """
+        EventTag = request.env['event.tag']
+        try:
+            tag_ids = list(filter(None, [request.env['ir.http']._unslug(tag)[1] for tag in (search_tags or '').split(',')]))
+        except Exception:
+            return EventTag
+
+        return EventTag.search([('id', 'in', tag_ids)]) if tag_ids else EventTag
```

Comment:
Since we're on the topic, after reviewing it again, it seems that searches['tags'] is always empty (though I'm not yet certain if it gets assigned a value elsewhere). It appears unused, so we might consider removing it, which could also eliminate the need for an additional function to handle it. What do you think? :)